policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceAsyncProxyTest.java

```diff
@@ -68,14 +68,23 @@ void startUp() {
     void testGetResourceRulesWithAPolicyRequest() throws Exception {
         // When
         when(hierarchyProxy.getResourceRules(any())).thenReturn(RESOURCE_RULES);
-        CompletableFuture<AuditablePolicyResourceRules> completableFuture = asyncProxy.getResourceRules(REQUEST);
-        AuditablePolicyResourceRules response = completableFuture.get();
+        AuditablePolicyResourceRules response = asyncProxy.getResourceRules(REQUEST).get();
 
         // Then
-        assertThat(response.getPolicyRequest()).isNotNull();
-        assertThat(response.getRules()).isNotNull();
-        assertThat(response.getAuditErrorMessage()).isNull();
         verify(hierarchyProxy, times(1)).getResourceRules(any());
+        assertAll("Check the values of the returned AuditablePolicyResourceRules object",
+                () -> assertThat(response.getPolicyRequest())
+                        .as("Check the PolicyRequest value is not null")
+                        .isNotNull(),
+
+                () -> assertThat(response.getRules())
+                        .as("Check the Rules value is not null")
+                        .isNotNull(),
```

Comment:
I'm not sure this does what the test wants. The assertions are being done on the elements of the returned auditable response and this recursive comparison is only being done on the response, not the request or rules elements. Taking out the `.ignoringFieldsOfTypes(AuditErrorMessage.class)` does not affect the outcome of the test.

Line 107 is checking that the elements inside the auditable response are null. It's not checking the returned response object as this will always be created by the Policy Service. When updating this test to follow the above assertion I get the following test error:
`[Check the PolicyRequest and the resource rules are null] 
Expecting:
 <AuditablePolicyResourceRules[policyRequest=null, rules=null, auditErrorMessage=null]>
to be equal to:
 <null>
but was not.
Comparison Failure: 
Expected :null
Actual   :AuditablePolicyResourceRules[policyRequest=null, rules=null, auditErrorMessage=null]`

I have left these assertions as they are for now.

Reply:
I've adjusted how the assertion is done so that a recursive comparison is done.